src/explorer/model/MavenProjectNode.ts

```diff
@@ -18,7 +18,8 @@ export class MavenProjectNode extends NodeBase {
     }
 
     public async getTreeItem(): Promise<vscode.TreeItem> {
-        await this._ensureMavenProjectParsed();
+        await this.ensureMavenProjectParsed();
```

Comment:
There is a potential NPE for `this._mavenProject.name` in next line. Although currently in this PR projects with invalid pom have been filtered out in `WorkspaceFolder.getChildren`, it's safer to check it here to avoid potential NPEs in future.

e.g. something like
```typescript
if (! await this.isValidPom()) {
    return undefined;
} 
```